File: src/main/java/main/utils/resume/ResumeData.java
package main.utils.resume;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.Getter;

import javax.annotation.Nullable;
import java.util.List;

public class ResumeData {
    private static final ObjectMapper mapper = new ObjectMapper();

    @Getter
    private final String channel_id;
    @Getter
    private final List<ResumableTrack> tracks;

    public ResumeData() {
        this.channel_id = null;
        this.tracks = null;
    }

    public ResumeData(String channel_id, List<ResumableTrack> tracks) {
        this.channel_id = channel_id;
        this.tracks = tracks;
    }

    @Override
    public String toString() {
        try {
            return mapper.writeValueAsString(this);
        } catch (JsonProcessingException e) {
            return null;
        }
    }

    public static ResumeData fromJSON(String json) throws JsonProcessingException {
        return mapper.readValue(json, ResumeData.class);
    }
}

File: src/main/kotlin/main/utils/resume/ResumeData.kt
//package main.utils.resume
//
//import kotlinx.serialization.Serializable
//import kotlinx.serialization.decodeFromString
//import kotlinx.serialization.encodeToString
//import kotlinx.serialization.json.Json
//
//@Serializable
//data class ResumeData(val channel_id: String, val tracks: List<ResumableTrack>) {
//
//    companion object {
//        fun fromJSON(json: String): ResumeData =
//            Json.decodeFromString(json)
//    }
//
//    override fun toString(): String =
//        Json.encodeToString(this)
//}
